### PR TITLE
- [google-cloud-cpp] bump abseil

### PR DESCRIPTION
**Pull request by SSE4**
closes: #9615 

Specify library name and version:  **google-cloud-cpp/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by DS-Serafin** [comment]
It does not completly close #9615 there sit still a clash on the protbuf version I think.

**Comment by DS-Serafin** [comment]
And openssl as well:

WARN: grpc/1.39.1: requirement openssl/1.1.1m overridden by google-cloud-cpp/1.31.1 to openssl/1.1.1l 
WARN: libcurl/7.78.0: requirement openssl/1.1.1m overridden by google-cloud-cpp/1.31.1 to openssl/1.1.1l

File: recipes/google-cloud-cpp/all/conanfile.py [hunk 1]
```diff
@@ -65,11 +65,11 @@ def source(self):
...
     def requirements(self):
-        self.requires('protobuf/3.17.1')
+        self.requires('protobuf/3.19.2')
         self.requires('grpc/1.39.1')
         self.requires('nlohmann_json/3.10.2')
         self.requires('crc32c/1.1.1')
-        self.requires('abseil/20210324.2')
+        self.requires('abseil/20211102.0')
         self.requires('libcurl/7.78.0')
         self.requires('openssl/1.1.1l')
```
Comment (SSE4) [inline recipes/google-cloud-cpp/all/conanfile.py:74]:
```suggestion
        self.requires('openssl/1.1.1m')
```

**Review by intelligide — APPROVED** [review]

**Comment by SSE4** [comment]
@DS-Serafin I'll take a look what's going on with gcc11 and let you know.
we have GCC11 builders, but it requires all dependencies to be built with GCC11 first. see an explanation: https://github.com/conan-io/conan-center-index/pull/8587

**Comment by SSE4** [comment]
> @SSE4 we have troubles building this with ggc 11 is there a reason the ci onl build up to gcc 10? Could you add a builder for gcc 11?

I've checked locally GCC11, and indeed it has some undefined references in abseil:
```
descriptor_utils.cc:(.text+0xc745): undefined reference to `absl::lts_20211102::variant_internal::ThrowBadVariantAccess()'
/usr/bin/ld: descriptor_utils.cc:(.text+0xc74a): undefined reference to `absl::lts_20211102::variant_internal::ThrowBadVariantAccess()'
/usr/bin/ld: descriptor_utils.cc:(.text+0xc74f): undefined reference to `absl::lts_20211102::variant_internal::ThrowBadVariantAccess()'
/usr/bin/ld: descriptor_utils.cc:(.text+0xc765): undefined reference to `absl::lts_20211102::variant_internal::ThrowBadVariantAccess()'
```
perhaps, an ABI mismatch/issue. anyway, as this PR is closed, please open new issue and let's discuss there.
meanwhile, I have scheduled dependencies to build for GCC11, so next PR to google-cloud-cpp should show the error.

**Pull request merged**